arraycontext/context.py

```diff
@@ -349,6 +351,29 @@ def clone(self):
             "setup-only" array context "leaks" into the application.
         """
 
+    def compile(self, f: Callable[[Any], Any],
+            inputs_like: Tuple[Union[Number, np.ndarray], ...]) -> Callable[
+                ..., Any]:
+        """Compiles *f* for repeated use on this array context. *f* is expected
+        to be a `pure function <https://en.wikipedia.org/wiki/Pure_function>`__
+        performing an array computation.
+
+        Control flow statements (``if``, ``while``) that might take different
+        paths depending on the data lead to undefined behavior and are illegal.
+        Any data-dependent control flow must be expressed via array functions,
+        such as ``actx.np.where``.
+
+        *f* may be called on placeholder data, to obtain a representation
+        of the computation performed, or it may be called as part of the actual
+        computation, on actual data. If *f* is called on placeholder data,
+        it may be called only once (or a few times).
+
+        :arg f: the function executing the computation.
+        :arg inputs_like: the input arguments to the function.
+        :return: a function with the same signature as *f*.
+        """
```

Comment:
@majosm Here are some docs on how `compile` is supposed to work. I haven't worked spelled out what `inputs_like` is supposed to be because I'm campaigning to get rid of it at https://github.com/inducer/arraycontext/pull/14/files#r651426611.

I'd welcome any feedback you might have.

Reply:
Thanks! That answers the questions I had about it. The documentation looks good to me.